core/templates/services/playthrough.service.ts

```diff
@@ -1,334 +1,359 @@
+        time_spent_state_in_msecs: {value: 1000 * timeSpentInStateSecs}
+      }));
+    if (this.misTracker !== null && this.cstTracker !== null) {
+      this.misTracker.recordStateTransition(destStateName);
+      this.cstTracker.recordStateTransition(destStateName);
+    }
+  }
+
+  recordExplorationQuitAction(
+      stateName: string, timeSpentInStateSecs: number): void {
+    if (!this.hasRecordingBegun() || this.hasRecordingFinished()) {
+      return;
+    }
+
+    this.recordedLearnerActions.push(
+      this.learnerActionObjectFactory.createNewExplorationQuitAction({
+        state_name: {value: stateName},
+        time_spent_in_state_in_msecs: {value: 1000 * timeSpentInStateSecs}
+      }));
+    if (this.playthroughStopwatch !== null && this.eqTracker !== null) {
+      this.playthroughDurationInSecs =
+       this.playthroughStopwatch.getTimeInSecs();
+      this.eqTracker.recordExplorationQuit(
+        stateName, this.playthroughDurationInSecs);
+    }
+  }
+
+  storePlaythrough(): void {
+    if (this.isRecordedPlaythroughHelpful()) {
+      const playthrough = this.createNewPlaythrough();
+      if (playthrough !== null) {
+        this.playthroughBackendApiService.
+          storePlaythroughAsync(playthrough, 1);
+      }
+    }
+  }
+
+  /**
+    * The ordering of checks in this method prioritizes the following types of
+    * playthroughs:
+    *    1. MultipleIncorrectSubmissionsIssue
+    *    2. CyclicStateTransitionsIssue
+    *    3. EarlyQuitIssue
+    *
+    * If none of the issue types have been discovered, returns null instead.
+    */
+  private createNewPlaythrough(): Playthrough | null {
+    if (this.explorationVersion !== null &&
+       this.recordedLearnerActions !== null) {
+      if (this.misTracker && this.misTracker.foundAnIssue()) {
+        return this.playthroughObjectFactory
+          .createNewMultipleIncorrectSubmissionsPlaythrough(
+            this.explorationId, this.explorationVersion,
+            this.misTracker.generateIssueCustomizationArgs(),
+            this.recordedLearnerActions);
+      } else if (this.cstTracker && this.cstTracker.foundAnIssue()) {
+        return this.playthroughObjectFactory
+          .createNewCyclicStateTransitionsPlaythrough(
+            this.explorationId, this.explorationVersion,
+            this.cstTracker.generateIssueCustomizationArgs(),
+            this.recordedLearnerActions);
+      } else if (this.eqTracker && this.eqTracker.foundAnIssue()) {
+        return this.playthroughObjectFactory
+          .createNewEarlyQuitPlaythrough(
+            this.explorationId, this.explorationVersion,
+            this.eqTracker.generateIssueCustomizationArgs(),
+            this.recordedLearnerActions);
+      }
+      return null;
+    }
+    return null;
+  }
+
+
+  private isPlaythroughRecordingEnabled(): boolean {
+    return (
+      this.explorationFeaturesService.isPlaythroughRecordingEnabled() &&
+      this.learnerIsInSamplePopulation === true);

```

Comment:
```suggestion
      this.explorationFeaturesService.isPlaythroughRecordingEnabled() &&
      this.learnerIsInSamplePopulation
    );

```

Reply:
Done

Reply:
It is giving error Type 'boolean | null' is not assignable to type 'boolean'.

  Type 'null' is not assignable to type 'boolean'.

Reply:
Can you post the whole error message?

Reply:
This was the error`boolean | null is not assignable to type 'boolean'. `